src/Microsoft.ML.Data/Transforms/ColumnBindingsBase.cs

```diff
@@ -15,7 +15,7 @@
 
 namespace Microsoft.ML.Data
 {
-    internal abstract class SourceNameColumnBase
+    public abstract class SourceNameColumnBase
     {
         [Argument(ArgumentType.AtMostOnce, HelpText = "Name of the new column", ShortName = "name")]
         public string Name;
```

Comment:
>Name [](start = 22, length = 4)

The names will have to change in accordance with the conclusion of #2064, but let's have agreement on the general approach first. (If we don't also want the entry-points to change yet, remember the `Name` property on the attribute.)